pallets/automation-time/src/types.rs

```diff
@@ -87,37 +87,18 @@ impl<AccountId: Clone + Decode, Balance: AtLeast32BitUnsigned, CurrencyId: Defau
 	}
 }
 
-#[derive(Debug, Encode, Decode, TypeInfo)]
-#[scale_info(skip_type_params(MaxExecutionTimes))]
-pub enum Schedule<MaxExecutionTimes: Get<u32>> {
```

Comment:
There is an argument to be made that encoding the MaxExecutionTimes in the type is a better implementation since its a stronger type bound and maybe less prone to error. However, in order to use the `Schedule` enum at the api boundary when scheduling tasks, the enum needs to be entirely serializable. `T::MaxExecutionTimes` does not currently fit that requirement and I don't think the work required to make it fit is worth the cost.

PR in question that adds `Schedule` to the api boundary- https://github.com/OAK-Foundation/OAK-blockchain/pull/300

Reply:
Yes I agree. The delta is not big enough for the serialization of enum work.